src/vueWrapper.ts

```diff
@@ -36,7 +36,12 @@ export class VueWrapper<T extends ComponentPublicInstance>
     this.__app = app
     // root is null on functional components
     this.rootVM = vm?.$root
-    this.componentVM = vm as T
+    // vm.$.proxy is what the template has access to
+    // so even if the component is closed (as they are by default for `script setup`)
+    // a test will still be able to do something like
+    // `expect(wrapper.vm.count).toBe(1)`
+    // (note that vm can be null for functional components, hence the condition)
+    this.componentVM = vm ? (vm.$.proxy as T) : (vm as T)
```

Comment:
we could also consider alternatives like:
- exposing the proxy on a different field, like `wrapper.proxy`, allowing to do `expect(wrapper.proxy.count).toBe(2)`, without changing `wrapper.vm`
- add a mounting option to expose everything:
```ts
const wrapper = mount(Hello, { expose: true });
expect(wrapper.vm.count).toBe(2); // vm would be proxy here
```
- only do this for script setup components, or for components that do not expose anything

Reply:
`expect(wrapper.proxy.count).toBe(2)` doesn't seem ideal - unless you know Vue and `script setup` very well, it'd be hard to understand (and even document). I think tests should be written the same way, regardless of whether you are using script setup, etc.

Reply:
Yeah I feel the same. I think testing different types of components should not change the way tests are written.

Reply:
I totally support @cexbrayat point - but being a devil's advocate - component implementation specific already leaks into VTU: for example you can't use `data()` option for class components, you won't be able to use `components` param with functional components, etc. I'm a bit concerned that we are doing exceptions here, although `script setup` is designed to be "blackboxed"